Orleans.Providers.MongoDB/Membership/Store/Multiple/MultipleMembershipCollection.cs

```diff
@@ -149,11 +148,8 @@ public async Task DeleteMembershipTableEntries(string deploymentId)
 
             await session.WithTransactionAsync(async (sessionHandle, ct) =>
             {
-                await Task.WhenAll(
-                    Collection.DeleteManyAsync(sessionHandle, x => x.DeploymentId == deploymentId),
-                    tableVersionCollection.DeleteAsync(sessionHandle, deploymentId)
-                );
-
+                await Collection.DeleteManyAsync(sessionHandle, x => x.DeploymentId == deploymentId);
```

Comment:
What's the benefit of serial vs concurrent task execution here and in the above code?

Reply:
I understand, and thanks for the detailed explanation, but I'm assuming the transaction number is only "advanced" at the creation of the transaction (`WithTransactionAsync`) but then all operations within that transaction share the same number...
This part of the library I inherited so I'm always hesitant to change what has been working, especially the parts that I haven't written myself. but it seems to me as a safe change, less likely to cause issues than multi threaded execution, and I doubt it will degrade the performance of the provider.